code/modules/client/preference_setup/general/04_equipment.dm

```diff
@@ -35,14 +35,14 @@
 	var/undies = get_undies()
 	var/gender_socks = get_gender_socks()
 	if(!get_key_by_value(undies, pref.underwear))
-		pref.underwear = undies[undies[1]]
+		pref.underwear = null
 	if(!get_key_by_value(undershirt_t, pref.undershirt))
 		pref.undershirt = undershirt_t[undershirt_t[1]]
 	if(!get_key_by_value(gender_socks, pref.socks))
-		pref.socks = gender_socks[gender_socks[1]]
-
+		pref.socks = null

```

Comment:
Inserting a sanitize_setup() inside the `if(. == TOPIC_REFRESH)` would run the proc directly after user input, and before redrawing the character.
I'll make a special case for undies and socks, then.